### ISSUE TITLE
Explore extension recommendation for pandas dataframes viewers

### ISSUE THREAD

**Issue by rebornix** [microsoft/vscode-jupyter#14315]

**Comment by contang0** [comment]
Dtale is easily the best. Would be awesome to see it embedded in VS code

https://github.com/man-group/dtale

**Comment by rebornix** [comment]
With the latest changes from https://github.com/microsoft/vscode-jupyter/pull/14545, we now allow 3rd party data viewers to specify which data types they support, and this will determine whether or not an Open button is displayed in the variable viewer row. If there are multiple data viewers for the same data type, a quick pick will appear and ask the user to select the data viewer they prefer.

We can also display a tooltip for data types that don't have data viewers available, with a tooltip that says "No installed extension can view this data type. Click here to find recommended extensions". However, since we are exploring a native variable explorer in the VS Code core, this is not implemented in the variable explorer shipped with Jupyter.

**Comment by DonJayamanne** [comment]
@rebornix Not sure this is ready to be exposed to 3rd parties just yet
Some points we might want to look at 
* The argument passed into the command is `IJUpyterVariable`, seems to be very complex (I would have expected we need to just provide the name and the type)
* If someone else is consuming this, wouldn't they need access to the API to get the details of the variable?

**Comment by rebornix** [comment]
> The argument passed into the command is IJUpyterVariable, seems to be very complex (I would have expected we need to just provide the name and the type)
> If someone else is consuming this, wouldn't they need access to the API to get the details of the variable?

@DonJayamanne good catch and I fully agree. Using `IJupyterVariable` is intermediate and most of the properties on it are not needed.

> shouldn't the arguments passed into the command (thus passed into 3rd party extension) go through a general API review/discussion?

Yep, if we are finalizing it, we go through API reviews. Also to fully build a data viewer extension, the `Kernel` API is needed, I'll bring this up when we are building the new Kernel API.
